### DISCUSSION TITLE
Support for safe collocation of files and folders

### DISCUSSION THREAD

**Discussion by BranchBotanist** [wasp-lang/wasp#2345] [Archived]
Wasp primarily uses a declarative approach for structuring apps, meaning that it automatically handles much of the deployment process based on the framework's patterns. Files like those in the public folder or configuration files that follow the typical Wasp project structure (such as main.wasp, Prisma schemas, and source code) are automatically included in the deployment. You don't need to specify them explicitly.

However, for files that fall outside the framework's expected structure but are still in your repository, you would indeed have to exclude them manually using .waspignore. This approach does introduce the possibility of human error when manually managing exclusions. Unfortunately, Wasp doesn't currently support a mode where it strictly includes only what's required for deployment (like some frameworks do) by default, unless those files are explicitly marked for inclusion.

The current safest way is to follow the Wasp project structure closely and use .waspignore for anything extra, but it doesn't yet provide a fully automated "safe inclusion" system based purely on default patterns

**Comment by Martinsos** [comment]
Thanks for pointing this out @BranchBotanist !
@infomiho @sodic does this actually happen? When we build the apps (client and server), would a file that is in the project, but is not used / imported by other files, be added to the final product? I guess at client it wouldn't due to Vite (do we we have it configured like that?) while on server it might? ALthough we do have a bundler on server side also if I remember correctly, so that might actually drop those files out then?

If not, what is the solution? Bunlding on both sides right?

**Comment by infomiho** [comment]
> Unfortunately, Wasp doesn't currently support a mode where it strictly includes only what's required for deployment (like some frameworks do) by default, unless those files are explicitly marked for inclusion.

@BranchBotanist could you give us an example to check out?

**Comment by Martinsos** [comment]
Moved this to an issue!

**Reply by sodic** [reply]
Link pls. Seems like GitHub didn't link it automatically

**Reply by Martinsos** [reply]
It did but it is on the right, quite small, hard to see. Here is the link! https://github.com/wasp-lang/wasp/issues/2356
